Dockerfile

```diff
@@ -1,22 +1,28 @@
 FROM nodesource/trusty:LTS
+
 MAINTAINER Jonathan Prince <jonathan.prince@gmail.com>
 
-RUN sed 's/main$/main universe/' -i /etc/apt/sources.list
-RUN apt-get update
-RUN apt-get upgrade -y
+# no tty
+ARG DEBIAN_FRONTEND=noninteractive
 
-# Download and install wkhtmltopdf
-RUN apt-get install -y build-essential xorg libssl-dev libxrender-dev wget gdebi
-RUN wget http://download.gna.org/wkhtmltopdf/0.12/0.12.2.1/wkhtmltox-0.12.2.1_linux-trusty-amd64.deb
-RUN gdebi --n wkhtmltox-0.12.2.1_linux-trusty-amd64.deb
+RUN sed 's/main$/main universe/' -i /etc/apt/sources.list
 
-# install pdftk
-RUN sudo apt-get install pdftk --yes
+# Install pdftk, download and install wkhtmltopdf
+RUN build_deps="build-essential xorg libssl-dev libxrender-dev wget gdebi" \
+  && apt-get update \
+  && apt-get install -y --force-yes --no-install-recommends $build_deps pdftk \
+  && wget http://download.gna.org/wkhtmltopdf/0.12/0.12.2.1/wkhtmltox-0.12.2.1_linux-trusty-amd64.deb \
+  && gdebi --n wkhtmltox-0.12.2.1_linux-trusty-amd64.deb \
+  && rm -f wkhtmltox-0.12.2.1_linux-trusty-amd64.deb \
+  && rm -rf /var/lib/apts/lists/* \
+  && apt-get purge -y --auto-remove $build_deps
 
-COPY ./ ./
+COPY package.json .
```

Comment:
Copy package.json separately so the `RUN` instruction creates a layer in Docker cache.